R/render-plot.R

```diff
@@ -173,7 +180,14 @@ renderPlot <- function(expr, width = 'auto', height = 'auto', res = 72, ...,
   outputFunc <- plotOutput
   if (!identical(height, 'auto')) formals(outputFunc)['height'] <- list(NULL)
 
-  markRenderFunction(outputFunc, renderFunc, outputArgs, cacheHint = FALSE)
+  markedFunc <- markRenderFunction(
+    outputFunc,
+    renderFunc,
+    outputArgs,
+    cacheHint = list(userExpr = get_expr(expr), res = res)
```

Comment:
I think we should add `outputName`, `altWrapper()`, and `args` here? Also, could change `args <- list(...)` to `args <- rlang::list2(...)`?

Reply:
- We don't use `outputName` in any other cache hints, so I don't want to add it here. This allows there to be multiple plots in a page that share the same cache, which could actually useful. (I know we use `outputName` in `renderCachedPlot`, but I think it's better to not use it.)
- If we pass `altWrapper`, we'll need to make a change so that the `cacheHint` is dynamically evaluated (and reactive). Currently the `cacheHint` is pre-hashed when `bindCache` is called, instead of every time the the key is invalidated. I think it makes sense to do this change.
- The `args <- list(...)` from `renderCachedPlot` doesn't actually happen in `bindCache`. We might want to expose that as a parameter (instead of using `...`) because those are already used for the key expression.

Reply:
Actually, now that I look at it, making the cache hint dynamic (for `altWrapper`) will add in enough complexity that I'd like to do it in a separate PR.

Currently, `cacheHint` just treated as a list of regular objects when it is hashed. If it is to be dynamic, then we may need to inspect each element to see if it's a quosure (or function, maybe?) and if so, evaluate it.